Frameworks/UIKit/UIPasteboard.mm

```diff
@@ -238,21 +344,23 @@ - (id)valueForPasteboardType:(NSString*)pasteboardType {
  @Notes only NSString and UIImage are supported.
 */
 - (void)setData:(NSData*)data forPasteboardType:(NSString*)pasteboardType {
-    if (data == nil || pasteboardType == nil) {
+    if (![_name isEqualToString:UIPasteboardNameGeneral]) {
+        UNIMPLEMENTED_WITH_MSG("Only general pasteboard supported");
         return;
     }
 
-    // Copy and Paste
-    if ([_name isEqualToString:UIPasteboardNameGeneral]) {
-        WADDataPackage* dataPackage = [[WADDataPackage make] autorelease];
-        dataPackage.requestedOperation = WADDataPackageOperationCopy;
+    if (data && pasteboardType) {
+        // only set data when both data and pasteType are avaiable
+        StrongId<WADDataPackage> dataPackage = [WADDataPackage make];
```

Comment:
sorry - i forgot to mention that I'm pretty sure you need to do a two-line init for this so you don't double addref (make inits the object, then the assignment adds an additional reference).  You (unfortunately) need to:

StrongId<WADDataPackage> dataPackage;
dataPackage.assign([WADDataPackage make]);

you can step into the debugger to confirm, but I'm pretty sure that's what's needed here (and anywhere else you're assigning to a newly-alloc'd object)